src/usecases/creditPageFunctions.ts

```diff
@@ -112,17 +117,9 @@ export const updateReactiveTags = <T extends CreditTag | DisplayTag>(
       reactiveTag.assign = compTag.assign;
   });
 
-  // この時点で reactiveTags に変更が行われたため、 reactiveTags と compTags の要素は一致しており、順番のみが異なる。
-
-  // update order
-  const idToOrder = compTags.reduce<Record<string, number>>(
-    (idToOrder, tag, i) => {
-      idToOrder[tag.id] = i;
-      return idToOrder;
-    },
-    {}
-  );
-  reactiveTags.sort((a, b) => idToOrder[a.id] - idToOrder[b.id]);
+  // add tags
+  const newTags = compTags.filter((tag) => newTagIds.includes(tag.id));
+  newTags.forEach((tag) => reactiveTags.push(tag));
```

Comment:
スプレッド構文で配列の変更がビューに反映されるかはわかりません。
配列のリアクティビティが保証されているのは以下のメソッドのみらしいので、`push`を使用しています。
- push()
- pop()
- shift()
- unshift()
- splice()
- sort()
- reverse()

https://jp.vuejs.org/v2/guide/list.html

for文を使わない書き方では`splice`を使う感じになるでしょうか。
```suggestion
  reactiveTags.splice(reactivetags.length, 0, ...newTags);
```

Reply:
そもそもなのですが layered architecture っぽい雰囲気で作られている Twinte-front において usecase が presentaion 層の問題である reactive に対して関心があるということが物事を難しくしている感があります。
ここらへんは一度しっかり決めてもいいのかもしれませんね。

※これはコードを変えてほしいという意図ではありません。これを気にしっかりアーキテクチャをチーム内で話したほうがいいかもねという提案です。

Reply:
`usecases/creditPageFunctions.ts`内にある`updateReactiveTags`と`updateCreditCourseWithStateList`は `reactive()`な値の変更を行うための処理なので、presentaion 層の`pages/credit.vue`に戻した方がわかりやすそうですね。

Reply:
アーキテクチャ関連の話で、
- `entity` でapi関連の型を用いた処理は書きたくない
- twinte-front における`usecase`の役割が曖昧 

だなと思っていた次第です。

Reply:
たしかにそうかもしれませんね。
ただこの場で即時直すべきというよりかは、このようなアーキテクチャの不整合があることは理解しつつ、今回は merge して、別の機会でしっかり話し合ってリファクタリングするとかでもいいと思います…！

reactive 意外にも DIP（依存性逆転の原則）とか entitiy の責務とかいろいろチーム内で考えたいな～と思うことがあり、この場で全て解決するのは難しいので。

Reply:
想定のアーキテクチャとしては、

- usecaseはフレームワーク非依存
- リアクティブなやつはhooksに置く

あたりが想定だった。
機能追加終わったらアーキテクチャのドキュメント化も含めてチーム内で話し合おう！